tests/test_ollama_provider.py

```
@@ -0,0 +1,348 @@
+class TestOllamaProviderErrorHandling:
+    """Test error handling and edge cases."""
+
+    @pytest.mark.asyncio
+    async def test_ollama_server_unavailable(self):
+        """Test graceful error when Ollama server is unavailable."""
+        # Use non-existent port
+        provider = OllamaProvider(base_url="http://localhost:99999")
+
+        request = LLMRequest(user_prompt="Test")
+
+        with pytest.raises(Exception):  # Should raise connection error
```

Ruff (0.14.4)

Do not assert blind exception: `Exception`

(B017)